Add a strict parsing feature (enabled by default)

elrnv opened the pull request:
When enabled, strict will return an error when an obj contains a command
that is not in the spec.

The motivation for this change is that obj is simple enough to allow
others to easily extend it with domain specific commands. For instance
the ARCsim cloth solver extends objs with face adjacency data.

It is beneficial for casual users and researchers to be able to parse
such third party obj files.

elrnv committed:
Add a strict parsing feature (enabled by default)

When enabled, strict will return an error when an obj contains a command
that is not in the spec.

The motivation for this change is that obj is simple enough to allow
others to easily extend it with domain specific commands. For instance
the ARCsim cloth solver extends objs with face adjacency data.

It is beneficial for casual users and researchers to be able to parse
such third party obj files.

kvark reviewed (commented):
This is useful, thank you!
My only concern is that this is a compile time feature, when in fact it could be a part of run-time configuration. Would you be open to refactoring this in a way that doesn't involve a compile time feature?

elrnv committed:
Switch to a runtime non-strict loading

Replace the compile time option for strict vs. non-strict loading to a
runtime option via a LoadConfig struct.

Add a test for checking that this works as expected.

kvark reviewed (approved):
Great, thank you!
